### ISSUE TITLE
[otel] integration should sync sampling decision to/from otel in the "parentless" case

### ISSUE THREAD

**Issue by costela** [getsentry/sentry-go#678]
## Summary

Currently, using something like the following seems to lead to unexpected results:
```go
	trace.NewTracerProvider(
		trace.WithSampler(
			trace.ParentBased(
				trace.TraceIDRatioBased(cfg.SampleRate),
			),
		),
		trace.WithSpanProcessor(sentryotel.NewSentrySpanProcessor()),
	)
```

The decision from the otel sampler is ignored by sentry, which then samples again based on its own logic, leaving the otel and sentry samplig decisions disconnected.

Using _only_ `sentry.ClientOptions.TracesSampler` is not an option for similar reasons: its decision does not get passed back to otel, so otel's `Span.IsSampled()` returns an unrelated value to sentry's decision (so that you end up - e.g. - using a traceID in exemplars and logs that never actually got sent to sentry, or the other way around).

The otel and sentry sides seem to agree in the case where a parent span could be found.

Am I holding it wrong? :sweat_smile: 

## Steps To Reproduce

Initialize otel with:
```go
trace.NewTracerProvider(
	trace.WithSampler(trace.AlwaysSample()),
	trace.WithSpanProcessor(sentryotel.NewSentrySpanProcessor()),
)
```

Initialize sentry with:
```go
sentry.ClientOptions{
	//...
	EnableTracing:    true,
	TracesSampleRate: 0.0,
	//...
}
```

All traces created with `otel.Tracer("").Start()` will have `span.IsSampled() == true`, but none of them will be sent to sentry.

Switching the to `NeverSample()`/`TracesSampleRate: 1.0` causes the opposite problem.

## Expected Behavior

Sentry should respect OTel's `span.IsSampled()` and only sample trances if and only if it is true.

(The other way around seems a bit more difficult to pull off?)

### SDK
 - `sentry-go` 0.22.0: 
 - Go version: 1.20
 - Using Go Modules? yes

### Sentry
 - Using hosted Sentry in sentry.io? yes

**Comment by costela** [comment]
Added a bit more context here: https://github.com/getsentry/sentry-go/pull/679#issuecomment-1689508580

The current state in that PR would imply the docs have to be updated:
- using otel+sentryhttp needs further options to sentryhttp
- respecting otel sampling requires `TracesSampleRate` to always be 0 (otherwise both sampling mechanisms may diverge)

**Comment by cleptric** [comment]
Closing as #679 is moving in another direction.
TL;DR is that we won't support mixing Sentry instrumentation with Otel instrumentation.
